packages/block-library/src/post-content/index.php

```diff
@@ -8,16 +8,18 @@
 /**
  * Renders the `core/post-content` block on the server.
  *
+ * @param WP_Block $block Block instance.
+ *
  * @return string Returns the filtered post content of the current post.
  */
-function render_block_core_post_content() {
-	$post = gutenberg_get_post_from_context();
-	if ( ! $post ) {
+function render_block_core_post_content( $block ) {
+	if ( ! isset( $block->context['postId'] ) ) {
 		return '';
 	}
+
 	return (
 		'<div class="entry-content">' .
-			apply_filters( 'the_content', str_replace( ']]>', ']]&gt;', get_the_content( $post ) ) ) .
+			apply_filters( 'the_content', str_replace( ']]>', ']]&gt;', get_the_content( null, false, $block->context['postId'] ) ) ) .
```

Comment:
Why not use this?
```suggestion
			apply_filters( 'the_content', str_replace( ']]>', ']]&gt;', get_post( $block->context['postId'] )->post_content ) ) .
```

Reply:
> Why not use this?

There's a few different ways it could be done (e.g. there's also [`get_post_field`](https://developer.wordpress.org/reference/functions/get_post_field/)). Noted in my original comment, I'm pretty intentional to limit the scope of the pull request to not include general refactoring. This specific logic required a bit more revisions only because the original implementation was _accidentally_ working, and didn't function correctly when replaced with block context.